jplag.frontend.swift/src/main/java/de/jplag/swift/grammar/SwiftSupport.java

```diff
@@ -0,0 +1,345 @@
+package de.jplag.swift.grammar;
+
+import java.util.BitSet;
+
+import org.antlr.v4.runtime.Token;
+import org.antlr.v4.runtime.TokenStream;
+import org.antlr.v4.runtime.misc.Interval;
+
+// Taken from https://github.com/antlr/grammars-v4/tree/master/swift/swift5 Commit
+// 2f6c19cc742c60541227b19c45ac2acc844d9b1a
+public class SwiftSupport {
+    /*
+     * TODO There is one caveat to the rules above. If the ! or ? predefined operator has no whitespace on the left, it is
+     * treated as a postfix operator, regardless of whether it has whitespace on the right. To use the ? as the
+     * optional-chaining operator, it must not have whitespace on the left. To use it in the ternary conditional (? :)
+     * operator, it must have whitespace around both sides.
+     */
+
+    /*
+     * operator-head : / = - + ! * % < > & | ^ ~ ? | [\u00A1-\u00A7] | [\u00A9\u00AB] | [\u00AC\u00AE] |
+     * [\u00B0-\u00B1\u00B6\u00BB\u00BF\u00D7\u00F7] | [\u2016-\u2017\u2020-\u2027] | [\u2030-\u203E] | [\u2041-\u2053] |
+     * [\u2055-\u205E] | [\u2190-\u23FF] | [\u2500-\u2775] | [\u2794-\u2BFF] | [\u2E00-\u2E7F] | [\u3001-\u3003] |
+     * [\u3008-\u3030] ;
+     */
+    public static final BitSet operatorHead = new BitSet(0x10000);
+    public static final BitSet operatorCharacter;
+
+    public static final BitSet leftWS = new BitSet(255);
+    public static final BitSet rightWS = new BitSet(255);
```

Comment:
What is "WS" ?

Reply:
Whitespace.
But as stated in the header of that file, it is taken from the ANTLR grammar repository. I can clean it up if required.